src/TransformerServiceProvider.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Surgiie\Transformer;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\ServiceProvider;
+
+class TransformerServiceProvider extends ServiceProvider
+{
+    public function boot()
+    {
+        Request::macro('transform', function (array $input, array $transformers = null): array {
+            $localTransformers = $transformers ?? $input;
+            $input = is_null($transformers)
+                ? (method_exists($this, 'validated')
```

Comment:
I think using `validated` if the request object is a `FormReques`t is okay but does have a silent omission of transformation if a validation rule is not defined in the form request.  Example: 

If you do something like: `$formRequest->transform(['name' => 'trim|ucfirst'])`

but dont have a rule defined in the form request, i.e:

```php
public function rules(){
    return [
        'name'=>['...'] 
    ];
}
```

That data is not transformed, and it will not be present in the transformed data array returned from `transform` which may cause some confusion, however I think im okay with this for now and just pointing it out, Ill make a note of this on the README in a future commit.

Reply:
Yes that's a valid point. It's a bit presumptuous.